### PR TITLE
Serverless Monitoring Guide

### PR DESCRIPTION
**Pull request by Hesperide**
PR for a serverless monitoring setup guide. Included for both Node.js and Python.

View on staging here: https://docs-staging.datadoghq.com/alex.cuoci/serverless-monitoring-guide/infrastructure/serverless/serverless_monitoring_guide/

### CODE REVIEW

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+# Section 2: Navigate Lambda Error Logs
+
+## Enable Lambda Log Ingestion
+
+In the previous section, we set up the Datadog Forwarder to enable enhanced metrics. The Datadog Forwarder is also required to send logs from your Lambda function to Datadog.
+
+{{< tabs >}}
+{{% tab "Serverless Framework" %}}
+
+If you’ve been following along, the Datadog Forwarder is already enabled, and Lambda logs should already automatically be flowing through to Datadog. If you have been following step-by-step but logs aren’t appearing in Datadog, [refer to the Lambda logs troubleshooting steps][1].
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:152]:
```suggestion
At this point, the Datadog Forwarder is enabled, and Lambda logs should already automatically be flowing through to Datadog. If  logs aren’t yet appearing in Datadog, [refer to the Lambda logs troubleshooting steps][1].
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+        # Add the ARN of your Forwarder.
+        forwarder:
+```
+
+[1]: https://docs.datadoghq.com/logs/guide/lambda-logs-collection-troubleshooting-guide/
+
+{{% /tab %}}
+{{% tab "AWS Console" %}}
+
+If you’ve been following along, the Datadog Forwarder is already enabled, and Lambda logs should already automatically be flowing through to Datadog. If you have been following step-by-step but logs aren’t appearing in Datadog, [refer to the Lambda logs troubleshooting steps][1].
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:172]:
```suggestion
At this point, the Datadog Forwarder is enabled, and Lambda logs should already automatically be flowing through to Datadog. If  logs aren’t yet appearing in Datadog, [refer to the Lambda logs troubleshooting steps][1].
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+[1]: https://docs.datadoghq.com/logs/guide/lambda-logs-collection-troubleshooting-guide/
+
+{{% /tab %}}
+{{% /tabs %}}
+
+If your logs still aren’t appearing in Datadog, you can [follow the troubleshooting steps here][9].  
+
+## Triage your Lambda Errors in the Log Explorer
+
+We can visit the Datadog Serverless view again to view all of our Lambda functions. Now, if we navigate to a specific function, we are able to see the logs being emitted from the function. From here, we can navigate to the Log Explorer to learn more about the log.
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:188]:
```suggestion
Visit the Datadog Serverless view to view all of the Lambda functions. Navigate to a specific function to see the logs being emitted from the function. Then navigate to the Log Explorer to learn more about the log.
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+# Section 3: Root Cause Analysis for Lambda Functions
+
+## Enable Lambda Distributed Tracing
+
+In the first section, we set up the Datadog Forwarder and Lambda Layer to enable enhanced metrics. Both the Datadog Forwarder and Lambda Layer are also required to surface distributed traces from your functions in Datadog.
+
+{{< tabs >}}
+{{% tab "Serverless Framework" %}}
+
+If you’ve been following along, the Datadog Forwarder and Lambda Layer are already enabled. With the Serverless Framework, traces are automatically flowing through to Datadog without any additional code instrumentation.
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:205]:
```suggestion
At this point, the Datadog Forwarder is and Lambda Layer are already enabled. With the Serverless Framework, traces are automatically flowing through to Datadog without any additional code instrumentation.
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+
+{{% /tab %}}
+{{% tab "Code Instrumentation" %}}
+
+If you’ve been following along, the Datadog Forwarder is enabled and the Lambda Layer has been installed on your function. Follow these steps to set up APM in Node:
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:210]:
```suggestion
Follow these steps to set up APM in Node:  
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+To instrument your Node libraries and customize your traces, consult the [Datadog Node APM documentation][1].
+
+[1]: https://docs.datadoghq.com/tracing/setup/nodejs/
+
+{{% /tab %}}
+{{% /tabs %}}
+
+## Root Cause Analysis using Traces, Metrics and Logs
+
+We can visit the Datadog Serverless page one more time to view all of our Lambda functions. If we navigate to a specific function, we are able to see the traces being emitted from the function. We can expand each trace to view a flame graph of the duration of the total request across Lambda functions, as well as correlated logs and Lambda metrics at the time of the request.
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:253]:
```suggestion
View the Datadog Serverless page to see the Lambda functions. Navigate to a specific function to see the traces being emitted from the function. Expand each trace to view a flame graph of the duration of the total request across Lambda functions, and correlated logs and Lambda metrics at the time of the request.
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+
+{{< img src="serverless/guides/end-to-end_serverless_monitoring_database/guide_video_5-compressed.mp4" video="true" alt="Flame Graph"  style="width:90%;">}}
+
+We can use the traces surfaced in the flame graph to identify that the current Lambda function has an error due to an error that surfaced from a subsequent function call. We can quickly navigate to that function in the request, and find the error trace and the corresponding error log, which we identify to be a timeout.
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:257]:
```suggestion
Use the traces surfaced in the flame graph to identify that the current Lambda function has an error due to an error that surfaced from a subsequent function call. Navigate to that function in the request, and find the error trace and the corresponding error log, which in this example is a timeout. 
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+
+We can visit the Datadog Serverless page one more time to view all of our Lambda functions. If we navigate to a specific function, we are able to see the traces being emitted from the function. We can expand each trace to view a flame graph of the duration of the total request across Lambda functions, as well as correlated logs and Lambda metrics at the time of the request.
+
+{{< img src="serverless/guides/end-to-end_serverless_monitoring_database/guide_video_5-compressed.mp4" video="true" alt="Flame Graph"  style="width:90%;">}}
+
+We can use the traces surfaced in the flame graph to identify that the current Lambda function has an error due to an error that surfaced from a subsequent function call. We can quickly navigate to that function in the request, and find the error trace and the corresponding error log, which we identify to be a timeout. 
+
+{{< img src="serverless/guides/end-to-end_serverless_monitoring_database/guide_video_6-compressed.mp4" video="true" alt="Root Cause Analysis"  style="width:90%;">}}
+
+With logs, traces and enhanced metrics all in one place, we are able to efficiently identify the root cause of an issue across our serverless requests.
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:261]:
```suggestion
With logs, traces, and enhanced metrics all in one place, we are able to efficiently identify the root cause of an issue across our serverless requests.
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+
+You can customize any of the default dashboards offered by the above integrations by pulling information from different services all into the same dashboard.
+
+## Submit Custom Metrics
+
+Custom metrics give additional insights into use cases that are unique to your application workflows, such as a user logging into your application, purchasing an item, or updating a user profile.
+
+In the first section, we set up the Datadog Lambda Layer to enable enhanced metrics. The Datadog Lambda Layer is required to surface distributed traces from your functions in Datadog.
+
+If you’ve been following along, the Datadog Lambda Layer is already enabled. Custom metrics sent from the Datadog Lambda Layer are automatically created as distributions, so you can graph the avg, sum, max, min, and count out-of-the-box. You can also enable your own percentiles in the Distribution Metrics page.
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:283]:
```suggestion
Once you have the Datadog Lambda Layer enabled, custom metrics are automatically created as distributions, so you can graph the `avg`, `sum`, `max`, `min`, and `count` out-of-the-box. You can also enable your own percentiles in the Distribution Metrics page.
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 10]
```diff
@@ -0,0 +1,359 @@
...
+        };
+}
+// You only need to wrap your function handler (Not helper functions).
+module.exports.myHandler = datadog(myHandler);
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+Creating a monitor for a metric enables you to get notified about key issues in your serverless applications. You could then determine root cause as illustrated earlier in this guide by combining metrics, traces and logs across your Lambda functions, data stores, and message queues.
```
Comment (apigirl) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:342]:
```suggestion
Creating a monitor for a metric enables you to get notified about key issues in your serverless applications. You could then determine root cause as illustrated earlier in this guide by combining metrics, traces, and logs across your Lambda functions, data stores, and message queues.
```

File: content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md [hunk 11]
```diff
@@ -0,0 +1,359 @@
...
+// You only need to wrap your function handler (Not helper functions).
+module.exports.myHandler = datadog(myHandler);
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+Creating a monitor for a metric enables you to get notified about key issues in your serverless applications. You could then determine root cause as illustrated earlier in this guide by combining metrics, traces, and logs across your Lambda functions, data stores, and message queues.
+
+{{< img src="serverless/guides/end-to-end_serverless_monitoring_database/guide_video_7-compressed.mp4" video="true" alt="Custom Metrics"  style="width:90%;">}}
```
Comment (pinkerton) [inline content/en/infrastructure/serverless/serverless_monitoring_guide/nodejs.md:344]: Similar feedback, let's reshoot this so we can see what's going on

**Pull request merged**